hydra-node/src/Hydra/Logging/Signaling.hs

```diff
@@ -0,0 +1,51 @@
+-- | A `Tracer` "middleware" to capture some logs and dispatch them to a `Server`.
+module Hydra.Logging.Signaling where
+
+import Hydra.Prelude
+
+import Control.Concurrent.Class.MonadSTM (newTVarIO, readTVarIO, writeTVar)
+import Hydra.API.Server (Server (..))
+import Hydra.API.ServerOutput (ServerOutput (SomeHeadInitializing))
+import Hydra.Chain.Direct.Handlers (DirectChainLog (SomeHeadObserved))
+import Hydra.Logging (Tracer (..))
+import Hydra.Logging.Messages (HydraLog (DirectChain))
+import Hydra.Cardano.Api (AssetName(..))
+import qualified Data.ByteString.Base16 as Hex
+
+withSignaling ::
+  (MonadSTM m) =>
+  Tracer m (HydraLog tx net) ->
+  (Tracer m (HydraLog tx net) -> TVar m (Maybe (Server tx m)) -> m a) ->
```

Comment:
Providing the TVar here on the callback is a bit odd and exposes internals. Instead, I would recommend returning only the `installSignal` callback function itself.

i.e. in usage:


```hs
withSignaling tracer $ \tracer' installSignal ->
  withChain tracer' ... $ \chain ->
    withServer tracer' chain ... $ \server -> 
      installSignal server
```

Also maybe even calling it `installSignalInServer server` in our usage site to make it clear that it's tying the knot between this signaling mechanism and the server.